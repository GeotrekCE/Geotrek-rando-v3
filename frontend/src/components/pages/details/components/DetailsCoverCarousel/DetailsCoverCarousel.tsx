import { Attachment } from 'modules/interface';
import { LargeCarousel } from 'components/Carousel';

interface DetailsCoverCarouselProps {
  attachments: Attachment[];
  onClickImage?: () => void;
}

export const DetailsCoverCarousel: React.FC<DetailsCoverCarouselProps> = ({
  attachments,
  onClickImage,
}) => {
  const files =
    typeof navigator !== 'undefined' && navigator?.onLine ? attachments : attachments.slice(0, 1);

  return (
    <LargeCarousel className="relative h-coverDetailsMobile desktop:h-coverDetailsDesktop">
      {files.map((attachment, i) => (
        <ImageWithLegend attachment={attachment} key={i} onClick={onClickImage} />
      ))}
    </LargeCarousel>
  );
};

interface ImageWithLegendProps {
  attachment: Attachment;
  onClick?: () => void;
}

export const ImageWithLegend: React.FC<ImageWithLegendProps> = ({ attachment, onClick }) => (
  <div
    id="details_cover_image"
    style={{ cursor: onClick ? 'pointer' : 'initial' }}
    className="relative"
    onClick={onClick}
  >
    <Legend author={attachment.author} legend={attachment.legend} />
    <img src={attachment.url} className="object-cover object-top overflow-hidden w-full h-full" />
  </div>
);

interface LegendProps {
  author: string;
  legend: string;
}

const Legend: React.FC<LegendProps> = ({ author, legend }) => {
  const hasLegendOrAuthor =
    (legend !== null && legend.length > 0) || (author !== null && author.length > 0);
  const hasLegendAndAuthor =
    legend !== null && legend.length > 0 && author !== null && author.length > 0;
  const fullText = `${legend}${hasLegendAndAuthor ? ' - ' : ''}${author}`;
  if (hasLegendOrAuthor) {
    return (
      hasLegendOrAuthor && (
        <div
          className={`w-full h-12 desktop:h-40
          absolute top-0 flex items-start justify-center
          py-1 px-2 pt-3 px-10
          bg-gradient-to-b from-blackSemiTransparent to-transparent
          text-white text-opacity-90 text-Mobile-C3 desktop:text-P2`}
        >
          <span className="mx-15 truncate">{fullText}</span>
        </div>
      )
    );
  }
  return null;
};
