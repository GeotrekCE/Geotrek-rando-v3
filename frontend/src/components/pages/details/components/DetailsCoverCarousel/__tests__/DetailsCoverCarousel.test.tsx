import { render } from 'services/testing/reactTestingLibraryWrapper';
import { DetailsCoverCarousel } from '../DetailsCoverCarousel';
describe('DetailsCoverCarousel', () => {
  it('should display a DetailsCoverCarousel', () => {
    const propsCoverCarousel = {
      attachments: [
        {
          url:
            'https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/lac_de_leychauda_parc_national_des_ecrins.jpg',
          legend: 'Lac',
          author: 'admin',
        },
        {
          url:
            'https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/le-depart-du-hameau-de-molines.JPG',
          legend: 'Marmotte',
          author: 'admin',
        },
        {
          url:
            'https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/arrivee-au-col-de-font-froide.JPG',
          legend: 'Arriv√©e au col de Font Froide',
          author: 'Dominique Vincent - PNE',
        },
      ],
    };
    const component = render(<DetailsCoverCarousel {...propsCoverCarousel} />);

    expect(component).toMatchSnapshot();
  });
});
