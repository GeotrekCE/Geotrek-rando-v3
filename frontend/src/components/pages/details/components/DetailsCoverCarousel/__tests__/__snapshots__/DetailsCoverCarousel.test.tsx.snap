// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCoverCarousel should display a DetailsCoverCarousel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="slick-slider sc-bcXHqe joHfNd relative h-coverDetailsMobile desktop:h-coverDetailsDesktop slick-initialized"
        dir="ltr"
      >
        <button
          class="sc-gswNZR sc-hLBbgP sc-iBYQkv dQACwy iXwfzt cBrSnO slick-arrow slick-prev"
          type="button"
        />
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-1"
              style="width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <figure
                  aria-labelledby=":r0:"
                  class="relative h-full "
                  id="details_cover_image"
                  role="figure"
                >
                  <img
                    alt="Lac"
                    class="object-center overflow-hidden size-full object-cover"
                    id=":r1:"
                    loading="lazy"
                    src="https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/lac_de_leychauda_parc_national_des_ecrins.jpg"
                  />
                  <figcaption
                    id=":r0:"
                  >
                    <span
                      class="w-full h-12 desktop:h-40
  absolute top-0 flex items-start justify-center
  pb-1 pt-3 px-2
  bg-gradient-to-b from-blackSemiTransparent to-transparent
  text-white text-opacity-90 text-Mobile-C3 desktop:text-P2"
                    >
                      <span
                        class="mx-10percent px-10percent truncate"
                      >
                        Lac - admin
                      </span>
                    </span>
                  </figcaption>
                </figure>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="3"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <figure
                  aria-labelledby=":r2:"
                  class="relative h-full "
                  id="details_cover_image"
                  role="figure"
                >
                  <img
                    alt="Lac"
                    class="object-center overflow-hidden size-full object-cover"
                    id=":r3:"
                    loading="lazy"
                    src="https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/lac_de_leychauda_parc_national_des_ecrins.jpg"
                  />
                  <figcaption
                    id=":r2:"
                  >
                    <span
                      class="w-full h-12 desktop:h-40
  absolute top-0 flex items-start justify-center
  pb-1 pt-3 px-2
  bg-gradient-to-b from-blackSemiTransparent to-transparent
  text-white text-opacity-90 text-Mobile-C3 desktop:text-P2"
                    >
                      <span
                        class="mx-10percent px-10percent truncate"
                      >
                        Lac - admin
                      </span>
                    </span>
                  </figcaption>
                </figure>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="4"
              style="width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="5"
              style="width: 0px;"
              tabindex="-1"
            />
          </div>
        </div>
        <button
          class="sc-gswNZR sc-hLBbgP sc-gKPRtg dQACwy iXwfzt bMoIHG slick-arrow slick-next"
          type="button"
        />
        <div
          class="sc-ftTHYK ktEDNK slick-dots"
        >
          <ul>
            <li
              class="slick-active"
            >
              <button>
                1
              </button>
            </li>
            <li
              class=""
            >
              <button>
                2
              </button>
            </li>
            <li
              class=""
            >
              <button>
                3
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="slick-slider sc-bcXHqe joHfNd relative h-coverDetailsMobile desktop:h-coverDetailsDesktop slick-initialized"
      dir="ltr"
    >
      <button
        class="sc-gswNZR sc-hLBbgP sc-iBYQkv dQACwy iXwfzt cBrSnO slick-arrow slick-prev"
        type="button"
      />
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          />
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <figure
                aria-labelledby=":r0:"
                class="relative h-full "
                id="details_cover_image"
                role="figure"
              >
                <img
                  alt="Lac"
                  class="object-center overflow-hidden size-full object-cover"
                  id=":r1:"
                  loading="lazy"
                  src="https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/lac_de_leychauda_parc_national_des_ecrins.jpg"
                />
                <figcaption
                  id=":r0:"
                >
                  <span
                    class="w-full h-12 desktop:h-40
  absolute top-0 flex items-start justify-center
  pb-1 pt-3 px-2
  bg-gradient-to-b from-blackSemiTransparent to-transparent
  text-white text-opacity-90 text-Mobile-C3 desktop:text-P2"
                  >
                    <span
                      class="mx-10percent px-10percent truncate"
                    >
                      Lac - admin
                    </span>
                  </span>
                </figcaption>
              </figure>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="3"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <figure
                aria-labelledby=":r2:"
                class="relative h-full "
                id="details_cover_image"
                role="figure"
              >
                <img
                  alt="Lac"
                  class="object-center overflow-hidden size-full object-cover"
                  id=":r3:"
                  loading="lazy"
                  src="https://geotrekdemo.ecrins-parcnational.fr/media/paperclip/trekking_trek/2/lac_de_leychauda_parc_national_des_ecrins.jpg"
                />
                <figcaption
                  id=":r2:"
                >
                  <span
                    class="w-full h-12 desktop:h-40
  absolute top-0 flex items-start justify-center
  pb-1 pt-3 px-2
  bg-gradient-to-b from-blackSemiTransparent to-transparent
  text-white text-opacity-90 text-Mobile-C3 desktop:text-P2"
                  >
                    <span
                      class="mx-10percent px-10percent truncate"
                    >
                      Lac - admin
                    </span>
                  </span>
                </figcaption>
              </figure>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="4"
            style="width: 0px;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width: 0px;"
            tabindex="-1"
          />
        </div>
      </div>
      <button
        class="sc-gswNZR sc-hLBbgP sc-gKPRtg dQACwy iXwfzt bMoIHG slick-arrow slick-next"
        type="button"
      />
      <div
        class="sc-ftTHYK ktEDNK slick-dots"
      >
        <ul>
          <li
            class="slick-active"
          >
            <button>
              1
            </button>
          </li>
          <li
            class=""
          >
            <button>
              2
            </button>
          </li>
          <li
            class=""
          >
            <button>
              3
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
